website/docs/r/api_access_key.html.markdown

```diff
@@ -64,3 +64,76 @@ For example:
 ```
 $ terraform import newrelic_api_access_key.foobar "1234567:INGEST"
 ```
+## Overview
+This module can be used for creation of api keys 
+
+### Input variables
+The following input variables are accepted by the module:
+
+* `api_key`: (Required) The NEW_RELIC ADMIN API key
+* `newrelic_account_id`: (Required) The New Relic account ID of the account you wish to create the API access key.
+* `key_type`: (Required) What type of API key to create. Valid options are INGEST or USER, case-sensitive.
+* `name`: (Optional) The name of the key.
+* `ingest_type`: (Optional) Required if key_type = INGEST. Valid options are BROWSER or LICENSE, case-sensitive.
+* `notes`: (Optional) Any notes about this ingest key.
+
+### Outputs
+The following output values are provided by the module:
+
+* `key`: The actual API key
+* `name`: The name of the key.
+* `type`: The type of API key
+
+
+### Example usage
+```terraform
+module "create_access_keys" {
+  source = "../examples/modules/fetch-newrelic-api-access-key"
+
+  create_access_keys_service = {
+    api_key = "NRAK-XXXXXXXXXX"
+    newrelic_account_id        = "12345678"
+    name                       = "key for CLI"
+    key_type                   = "INGEST"
+  }
+}
+
+output "required_attributes" {
+  value = module.create_access_keys.required_attributes
+}
+```
```

Comment:
@vinay-newrelic please format all examples used here, using the `terraform fmt` command. 

Additionally 
- please add examples for the three kinds of keys (USER, INGEST LICENSE and INGEST BROWSER)
- we do not need to explain what each argument does, so the Input Variables section is not required
- the `Outputs` section may be updated based on a previous comment of mine, if the `ingest_type` is also included
- with six examples added, since this file would turn out to be too large, I feel it would be a better idea to add all of this content in a `README.md` file in the module (on GitHub) and simply add a link to this in the docs of the `newrelic_api_access_key` resource (in alignment with one of your previous suggestions) :)

Reply:
@pranav-new-relic sounds good .Made the necessary changes as per our agreement then